internal/jobcontainers/jobcontainer.go

```diff
@@ -0,0 +1,393 @@
+
+	if s == nil {
+		return nil, errors.New("Spec must be supplied")
+	}
+
+	if id == "" {
+		g, err := guid.NewV4()
+		if err != nil {
+			return nil, err
+		}
+		id = g.String()
+	}
+
+	if err := mountLayers(ctx, s); err != nil {
+		return nil, errors.Wrap(err, "failed to mount container layers")
+	}
+
+	const volumeGUIDRegex = `^\\\\\?\\(Volume)\{{0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}\}(|\\)$`
+	if matched, err := regexp.MatchString(volumeGUIDRegex, s.Root.Path); !matched || err != nil {
+		return nil, fmt.Errorf(`invalid container spec - Root.Path '%s' must be a volume GUID path in the format '\\?\Volume{GUID}\'`, s.Root.Path)
+	}
+	if s.Root.Path[len(s.Root.Path)-1] != '\\' {
+		s.Root.Path += `\` // Be nice to clients and make sure well-formed for back-compat
+	}
+
+	container := newJobContainer(id, s)
+
+	// Create the job object all processes will run in. We set true for the notifications argument
+	// as we always want to receive messages from the job.
+	job, err := jobobject.Create(ctx, id, true)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create job object")
+	}
+	container.job = job
+
+	var path string
+	defer func() {
+		if err != nil {
+			container.Close()
+			if path != "" {
+				removeSandboxMountPoint(ctx, path)
+			}
+		}
+	}()
+
+	limits, err := specToLimits(ctx, s)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to convert OCI spec to job object limits")
+	}
+
+	// Set resource limits on the job object based off of oci spec.
+	if err := job.SetResourceLimits(limits); err != nil {
+		return nil, errors.Wrap(err, "failed to set resource limits")
+	}
+
+	// Setup directory sandbox volume will be mounted
+	sandboxPath := fmt.Sprintf(sandboxMountFormat, id)
+	if _, err := os.Stat(sandboxPath); err != nil {
+		if err := os.MkdirAll(sandboxPath, 0777); err != nil {
+			return nil, errors.Wrap(err, "failed to create mounted folder")
+		}
+	}
+	path = sandboxPath
+
+	if err := mountSandboxVolume(ctx, path, s.Root.Path); err != nil {
+		return nil, errors.Wrap(err, "failed to bind payload directory on host")
+	}
+
+	container.sandboxMount = path
+	go container.waitBackground(ctx)
+	return container, nil
+}
+
+// CreateProcess creates a process on the host, starts it, adds it to the containers
+// job object and then waits for exit.
+func (c *JobContainer) CreateProcess(ctx context.Context, config interface{}) (_ cow.Process, err error) {
+	conf, ok := config.(*hcsschema.ProcessParameters)
+	if !ok {
+		return nil, errors.New("unsupported process config passed in")
+	}
+
+	path, args := seperateArgs(conf.CommandLine)
+	absPath, err := findExecutable(path, c.sandboxMount)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find executable")
+	}
+
+	token, err := processToken(conf.User)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create process token")
+	}
+
+	cmd := &exec.Cmd{
+		Env:  os.Environ(),
```

Comment:
Should we be using or appending the environment passed in with the conf?

Reply:
Yes, and this also reminds me that it's missing explicitly adding in the path to the sandbox mount here so thank you. This was mostly just cped from the old privileged containers PR.